spark3/src/test/java/org/apache/iceberg/actions/TestCreateActions.java

```diff
@@ -157,6 +168,105 @@ public void testMigratePartitioned() throws Exception {
     assertMigratedFileCount(Actions.migrate(source), source, dest);
   }
 
+  @Test
+  public void testPartitionedTableWithUnRecoveredPartitions() throws Exception {
+    Assume.assumeTrue("Cannot migrate to a hadoop based catalog", !type.equals("hadoop"));
+    Assume.assumeTrue("Can only migrate from Spark Session Catalog", catalog.name().equals("spark_catalog"));
+    String source = sourceName("test_unrecovered_partitions");
+    String dest = source;
+    File location = temp.newFolder();
+    spark.sql(String.format(CREATE_PARTITIONED_PARQUET, source, location));
+    spark.range(5)
```

Comment:
nit: I usually prefer to have an empty line between different logical blocks. That way, it is easier to read.